pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1094,6 +1050,53 @@ func (c *K8sOrchestrator) GetAllK8sVolumes() []string {
 	return volumeIDs
 }
 
+// SetWcpCapabilitiesMap reads the capabilities values from 'supervisor-capabilities' CR in
+// supervisor cluster and sets values in global wcp capabilities map.
+func SetWcpCapabilitiesMap(ctx context.Context, cnsOperatorClient client.Client) error {
+	log := logger.GetLogger(ctx)
+	wcpCapabilitiesMutex.Lock()
+	defer wcpCapabilitiesMutex.Unlock()
+	// Check the 'supervisor-capabilities' CR in supervisor
+	wcpCapabilities := &wcpcapv1alph1.Capabilities{}
+	err := cnsOperatorClient.Get(ctx, k8stypes.NamespacedName{
+		Name: common.WCPCapabilitiesCRName},
+		wcpCapabilities)
+	if err != nil {
+		log.Errorf("failed to fetch Capabilities CR instance "+
+			" with name %q Error: %+v", common.WCPCapabilitiesCRName, err)
+		return err
+	}
+	if WcpCapabilitiesMap == nil {
+		WcpCapabilitiesMap = make(map[string]bool)
+	}
+	for capName, capStatus := range wcpCapabilities.Status.Supervisor {
+		WcpCapabilitiesMap[string(capName)] = capStatus.Activated
+	}
+	return nil
+}
+
+// setCapabilitiesMapOnSupervisor creates a new cns-operator client for supervisor cluster and
+// calls SetWcpCapabilitiesMap function to set latest capabilities values in global wcp capabilities map.
+func setCapabilitiesMapOnSupervisor(ctx context.Context) error {
+	log := logger.GetLogger(ctx)
+	restConfig, err := clientconfig.GetConfig()
+	if err != nil {
+		log.Errorf("failed to get Kubernetes config. Err: %+v", err)
+		return err
+	}
+	cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, wcpcapapis.GroupName)
+	if err != nil {
+		log.Errorf("failed to create CnsOperator client. Err: %+v", err)
```

Comment:
Update the log to indicate a failure to create wcpCapClient

Reply:
done